### PR TITLE
generic credential item storage

### PR DESCRIPTION
**Pull request by vijayshiyani**
Modify the credential vault item creation and retrieval process to be adaptable for various credential `formats`. 
This change will shift the responsibility of specifying the `format` and `details` of credentials to the user, allowing them to provide and store credentials in any desired format.

### CODE REVIEW

File: packages/sdk/src/util/constants.ts [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-}
-
 /**
  * Credential item attributes
  */
 export const CREDENTIAL_ITEM = {
   TEMPLATE_NAME: 'verifiable_credential',
-  JWT_SLOT_NAME: 'credential_jwt',
-  JWT_SLOT_LABEL: 'Credential JWT',
+  // TODO: rename to credential from credential_jwt as it may not be always a JWT - depends on the format https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#appendix-E.1.1.6
```
Comment (vijayshiyani) [inline packages/sdk/src/util/constants.ts:6]: @linasi 
I am guessing 
Implementing this change should not be too difficult, as it primarily involves updating the template and performing a migration of existing items. let me know your thoughts.

File: packages/sdk/src/services/credential-service.ts [hunk 2]
```diff
@@ -22,8 +21,21 @@ export interface GenerateCredentialExtendedDto extends Omit<GenerateCredentialDt
   issuer: string | { id: string; name?: string };
 }
 
+export interface CredentialDetail {
+  issuer: any | null;
+  subject: string | null;
+  issuanceDate: Date | null;
+  expirationDate: Date | null;
+  id: string | null;
+  credentialSchema: any | null;
```
Comment (linasi) [inline packages/sdk/src/services/credential-service.ts]: @vijayshiyani this should not be just `any`. When assigning value to the slot, it expects object with an `id` property: `{id: string}`;

File: packages/sdk/src/services/credential-service.ts [hunk 3]
```diff
@@ -22,8 +21,21 @@ export interface GenerateCredentialExtendedDto extends Omit<GenerateCredentialDt
   issuer: string | { id: string; name?: string };
 }
 
+export interface CredentialDetail {
+  issuer: any | null;
```
Comment (linasi) [inline packages/sdk/src/services/credential-service.ts]: @vijayshiyani same here, `any` should not be used. It is either `string` or object with an `id` property `{id: string}`

File: packages/sdk/src/services/credential-service.ts [hunk 4]
```diff
@@ -22,8 +21,21 @@ export interface GenerateCredentialExtendedDto extends Omit<GenerateCredentialDt
...
+  id: string | null;
+  credentialSchema: any | null;
+  revocable: boolean | null;
+}
+
 export interface CreateVerifiableCredentialItemParams {
-  credentialJWT: string;
+  credential: string;
+  format: string;
+  id: string;
```
Comment (linasi) [inline packages/sdk/src/services/credential-service.ts:37]: what is the difference with this `id` and `id` inside the `CredentialDetail`?

File: packages/sdk/test/services/credential-service.test.ts [hunk 5]
```diff
@@ -862,7 +1130,7 @@ describe('CredentialService', () => {
               },
             ],
             classification_nodes: [],
-            label: 'urn:uuid:66cfde0a-bac9-4ed3-8cee-10e4a417dcc9',
```
Comment (linasi) [inline packages/sdk/test/services/credential-service.test.ts:865]: why label has changed to the value without `urn:uuid`?

**Review by linasi — APPROVED** [review]
LGTM. Please take a look at those few comments I've left and make sure it works as expected if credential ID is `urn:uuid` or just whatever other random value.

File: packages/sdk/src/services/credential-service.ts (hunk 2 above)
Comment (vijayshiyani) [reply packages/sdk/src/services/credential-service.ts]: @linasi fix typings 
```
export interface CredentialDetail {
  issuer?: string | { id: string; name?: string };
  subject?: string;
  issuanceDate?: Date;
  expirationDate?: Date;
  id?: string;
  credentialSchema?: { id: string; type: string };
  revocable?: boolean;
}
```

File: packages/sdk/src/services/credential-service.ts (hunk 3 above)
Comment (vijayshiyani) [reply packages/sdk/src/services/credential-service.ts]: same as here  https://github.com/Meeco/js-sdk/pull/279/files#r1457138459

File: packages/sdk/src/services/credential-service.ts (hunk 4 above)
Comment (vijayshiyani) [reply packages/sdk/src/services/credential-service.ts:37]: @linasi 

`id` inside `CredentialDetail` refers to `vc-id` and it's optional not every `VC` will have `ID`
e.g.
```
"vc": {
    "id": "http://example.edu/credentials/3732",
..
    "credentialSubject": {
      "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
      "degree": {
        "type": "BachelorDegree",
        "name": "Bachelor of Science and Arts"
      }
    }
  },
```

The 'id' supplied for the function call `createVerifiableCredentialItem` is necessary and will be utilized as the `label` and `name` for the `vault-item`.

File: packages/sdk/test/services/credential-service.test.ts (hunk 5 above)
Comment (vijayshiyani) [reply packages/sdk/test/services/credential-service.test.ts:865]: @linasi 
Initially, I removed the `formatIdToItemName` function, thinking the client would provide the correct ID. However, upon reflection, I've decided to bring back the function to handle ID formatting within the SDK.

**Pull request merged**